### DISCUSSION TITLE
Is it possible to have models with same name but on different files when autogenerating migrations?

### DISCUSSION THREAD

**Discussion by oaltun** [sqlalchemy/alembic#988] [Usage Questions]
E.g. 

In `app.support.models.py`
```
class Status(Base):
    __tablename__: "support_status"
    ...
```

And in `app.blog.models.py`

```
class Status(Base):
    __tablename__: "blog_status"
    ...
```

Notice that the table names are different, so there should be no problem on the database side. 

I think this can be needed for large applications, also for apps that want to support things like plugins/apps/namespaces/blueprints. But I was not able to make it work.

At first I tried to approach like adding following in `alembic/env.py`:
```
from app.support.models import Base as SupportBase
from app.blog.models import Base as BlogBase
target_metadata = [SupportBase.metadata, BlogBase.metadata]
```
But I get an error about "duplicate keys" when autogenerating migrations. Applying existing migrations works fine, but I can not autogenerate. The documentation says keys are a combination of schema and tablenames. I dont know what a schema is, but tablenames are different here.

At this point to use separate *.model.py files I am importing all *.models.py files into a main models.py and use that in env.py.
E.g. in `app.models.py`:
```
from app.support.models import *
from app.blog.models import *
```
and in `alembic/env.py`:
```
from app.models import Base
target_metadata = Base.metadata
```
This works for using different models.py files for each section of my application, but it squashes any chances of using any model names in more than one files.

**Comment by CaselIT** [comment]
Hi,

I think the issue here is different. Alembic has no knowledge of the models, it only deals with tables.

In the example you have, is the `Base` in both files the same one? Or are you working with two bases? If it's the same one, then the metadata object is also the same and you should list it only once
